mlflow/utils/validation.py

```diff
@@ -52,14 +64,32 @@ def _validate_metric(key, value, timestamp):
             "double (64-bit floating point)" % (value, key, timestamp),
             INVALID_PARAMETER_VALUE)
 
-    if not isinstance(timestamp, numbers.Number) or timestamp < 0 or \
-            timestamp < np.iinfo(np.int64).min:
+    if not isinstance(timestamp, numbers.Number) or timestamp < 0:
         raise MlflowException(
             "Got invalid timestamp %s for metric '%s' (value=%s). Timestamp must be a nonnegative "
             "long (64-bit integer) " % (timestamp, key, value),
             INVALID_PARAMETER_VALUE)
 
 
+def _validate_param(key, value):
+    """
+    Check that a param with the specified key & value is valid and raise an exception if it
+    isn't.
+    """
+    _validate_param_name(key)
+    _validate_length_limit("Param key", MAX_ENTITY_KEY_LENGTH, key)
+    _validate_length_limit("Param value", MAX_PARAM_VAL_LENGTH, value)
+
+
+def _validate_tag(key, value):
+    """
+    Check that a tag with the specified key & value is valid and raise an exception if it isn't.
+    """
+    _validate_tag_name(key)
+    _validate_length_limit("Param key", MAX_ENTITY_KEY_LENGTH, key)
+    _validate_length_limit("Tag value", MAX_TAG_VAL_LENGTH, value)
+
+
 def _validate_param_name(name):
```

Comment:
Note: We may eventually want to deprecate the _validate_param_name, _validate_metric_name etc helpers - they’re only needed right now to validate names on the read path for the FileStore, so we could delete them entirely if/when we deprecate the FileStore.